Transform UI Updates and Pane Grid Structure

nofurtherinformation opened the pull request:
## Overview

This PR substantially revises the transform editor UI to be more intuitive and ergonomic. Additionally, this PR adds Grid css formatting to pane layouts, especially for overlay buttons and draggable interactions.

### Demo

<img width="1440" alt="Screen Shot 2022-10-25 at 10 17 59 AM" src="https://user-images.githubusercontent.com/10041771/197813870-95858bbe-e136-4892-942c-887eb9b2a360.png">
<img width="378" alt="Screen Shot 2022-10-25 at 10 19 15 AM" src="https://user-images.githubusercontent.com/10041771/197814172-0a1526a9-7dd0-43a1-8c94-3ad8c79dc429.png">

## Testing Instructions

* Add dataset transform
* Use panes, drag

nofurtherinformation commented:
The table can now be maximized/minimized with the arrow button. By default and on remount, it is minimized
<img width="1440" alt="Screen Shot 2022-10-25 at 10 34 38 AM" src="https://user-images.githubusercontent.com/10041771/197817922-6cf968ca-499a-48ea-9523-8a51b46acf0c.png">

stuartlynn commented:
This is looking good. One small thing I notices was that when I expand the table, it doesn't seem to get get bigger, just become dominant on the screen. Wonder if we just need to load more data and filter it down based on the zoom level?

![image](https://user-images.githubusercontent.com/209838/198639339-34b3193d-60f9-4b5d-b632-e3238bd4b3cb.png)

nofurtherinformation commented:
OK! Just implemented more rows for the data preview table on expanded. 

I also added the pill-style editor for category filters, which took a little finessing to get the interaction and a11y right, but I think we're good. 

<img width="458" alt="Screen Shot 2022-10-29 at 10 46 26 PM" src="https://user-images.githubusercontent.com/10041771/198862781-942ce5f5-6298-4282-b788-cb0d4e0d2712.png">

Also, also, I did a big ol refactor of the parts here, since the component files were getting pretty chunky!
